programs/stake/src/stake_state.rs

```diff
@@ -766,13 +770,31 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
                         return Err(InstructionError::InsufficientFunds);
                     }
                     // split the stake, subtract rent_exempt_balance unless
-                    //  the destination account already has those lamports
-                    //  in place.
-                    // this could represent a small loss of staked lamports
-                    //  if the split account starts out with a zero balance
-                    let split_stake = stake.split(
-                        lamports - meta.rent_exempt_reserve.saturating_sub(split.lamports()?),
-                    )?;
+                    // the destination account already has those lamports
+                    // in place.
+                    // this means that the new stake account will have a stake equivalent to
+                    // lamports minus rent_exempt_reserve if it starts out with a zero balance
+                    let (remaining_stake_delta, split_stake_amount) = if lamports
+                        == self.lamports()?
+                    {
+                        // If split amount equals the full source stake, the new split stake must
+                        // equal the same amount, regardless of any current lamport balance in the
+                        // split account. Since split accounts retain the state of their source
+                        // account, this prevents any magic activation of stake by prefunding the
+                        // split account.
+                        (
+                            lamports - meta.rent_exempt_reserve,
+                            lamports - meta.rent_exempt_reserve,
+                        )
+                    } else {
+                        // Otherwise, the new split stake should reflect the entire split
+                        // requested, less any lamports needed to cover the rent-exempt reserve
+                        (
+                            lamports,
+                            lamports - meta.rent_exempt_reserve.saturating_sub(split.lamports()?),
+                        )
+                    };
+                    let split_stake = stake.split(remaining_stake_delta, split_stake_amount)?;
 
                     self.set_state(&StakeState::Stake(meta, stake))?;
                     split.set_state(&StakeState::Stake(meta, split_stake))?;
```

Comment:
@CriesofCarrots oops, I found another issue. Namely, the split account's rent-reserve can be set incorrectly. I'm still reading the code, but we could never safely `set_state` identical  `meta` to two stake accounts. We have to individually maintain rent-exempt reserves. another head-ache...

Very unfortunately, the are various sizes of stake accounts on the wild:

```
$ ./target/release/solana stake-account C75NM6yJMa4uAEfdUYnSmMHfVAUiPVd67rqTKaZdd5Z
Balance: 500000 SOL
Rent Exempt Reserve: 0.00228288 SOL
Delegated Stake: 499999.99771712 SOL
Active Stake: 499999.99771712 SOL
Delegated Vote Account Address: CCxSNvJogH6LWyoiEbG7JfcWybw2FqqCExs5GuemChGr
Stake Authority: so1b2w9fpMdqgzHR2UvW4iqkkc8nig6xTeBjU5HMbjG
Withdraw Authority: FdGYQdiRky8NZzN9wZtczTBcWLYYRXrJ3LMDhqDPn5rM


$ ./target/release/solana stake-account 13CJic1hdT9woLL7ctrKi9rBwBFoqjcF2BNnDbZceDnS
Balance: 25 SOL
Rent Exempt Reserve: 0.02878656 SOL
Delegated Stake: 24.97121344 SOL
Active Stake: 24.97121344 SOL
Delegated Vote Account Address: ateamaZDqNWDztxnVKZhRsp4ac53KvT1rVKyU5LnL6o
Stake Authority: 55c8bMTU311mv8KkAnTMtdGGKWAueAGhNUYdVfvjuvNy
Withdraw Authority: 55c8bMTU311mv8KkAnTMtdGGKWAueAGhNUYdVfvjuvNy
```

Reply:
```
$ ./target/release/solana account C75NM6yJMa4uAEfdUYnSmMHfVAUiPVd67rqTKaZdd5Z | head

Public Key: C75NM6yJMa4uAEfdUYnSmMHfVAUiPVd67rqTKaZdd5Z
Balance: 500000 SOL
Owner: Stake11111111111111111111111111111111111111
Executable: false
Rent Epoch: 107
Length: 200 (0xc8) bytes
0000:   02 00 00 00  80 d5 22 00  00 00 00 00  0d 03 10 73   ......"........s
0010:   44 03 0b f0  4d b2 8a 86  2c 2d 19 93  1f fb c5 e1   D...M...,-......
0020:   50 36 55 ed  9f 03 88 e8  d0 d7 98 bb  d9 4e 19 53   P6U..........N.S

$ ./target/release/solana account 13CJic1hdT9woLL7ctrKi9rBwBFoqjcF2BNnDbZceDnS | head

Public Key: 13CJic1hdT9woLL7ctrKi9rBwBFoqjcF2BNnDbZceDnS
Balance: 25 SOL
Owner: Stake11111111111111111111111111111111111111
Executable: false
Rent Epoch: 107
Length: 4008 (0xfa8) bytes
0000:   02 00 00 00  80 3f b7 01  00 00 00 00  3c 9c cb 27   .....?......<..'
0010:   f9 18 85 f0  46 c2 c1 29  91 56 ab 80  e1 37 8f 30   ....F..).V...7.0
0020:   76 d8 fd ff  ab d7 d0 4b  38 d7 fd ce  3c 9c cb 27   v......K8...<..'
```

Reply:
That means my upcoming stake account normalization must also adjust rent-exempt reserves.

Reply:
ref: https://discord.com/channels/428295358100013066/439194979856809985/771378185890693131